src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs

```diff
@@ -1221,6 +1256,29 @@ private void InitializeCommands()
             UpdatePythonPathCommand = new DelegateCommand(p => UpdatePathAt());
         }
 
+        public bool UpdateBackupLocation(string newBackupLocation)
+        {
+            var previousBackupLocation = BackupLocation;
+            BackupLocation = newBackupLocation;
+
+            if (dynamoViewModel.Model.UpdateBackupLocation(newBackupLocation))
+            {
+                CanResetBackupLocation = !dynamoViewModel.Model.IsDefaultBackupLocation();
```

Comment:
if `CanResetBackupLocation = !dynamoViewModel.Model.IsDefaultBackupLocation();` is always valid, I would put it in the getter and remove the setter for that property

Reply:
It's if true if the user select a different location than the default, is false if the user reset the location to its default value.

Reply:
> if `CanResetBackupLocation = !dynamoViewModel.Model.IsDefaultBackupLocation();` is always valid, I would put it in the getter and remove the setter for that property. Looks like you dont have to set it to a different value here

It's if true if the user select a different location than the default, is false if the user reset the location to its default value.

Reply:
Understood, my point is that it looks like you are setting this same `!dynamoViewModel.Model.IsDefaultBackupLocation();` to it on different places, which implicate that it could just be its getter

Reply:
Based on the result of the UpdateBackupLocation true if it's ok / false if the path is invalid, I need to set the value of CanResetBackupLocation because it enables or not the  [Reset location button].

Reply:
refactored

Reply:
> yeah @jesusalvino In this case, do we still need the setter? usually only a getter here would be sufficient

yes @QilongTang , I have just tested applying the Binding Mode and UpdateSourceTrigger in the xaml side with success, the explicit setter is necessary due to is the only way to raise the property change in order to enable or not the [Reset button] in the view immediately after the user selects a Backup folder

Reply:
Thank you @jesusalvino I missed that point. In this case, feel free to revert the getter to previous change to align with setter. sorry for dragging the conversation. Looks good

Reply:
Done @QilongTang